zebra-chain/src/transaction/unmined/zip317.rs

```diff
@@ -28,5 +31,59 @@ const P2PKH_STANDARD_OUTPUT_SIZE: usize = 34;
 ///
 /// [ZIP-317]: https://zips.z.cash/zip-0317#fee-calculation
 pub fn conventional_fee(transaction: &Transaction) -> Amount<NonNegative> {
-    todo!()
+    // zcash_primitives checks for non-p2pkh inputs, but Zebra doesn't.
+    // Conventional fees are only used in the standard rules for mempool eviction
+    // and block production, so these implementations are compatible.
+    //
+    // <https://github.com/zcash/librustzcash/blob/main/zcash_primitives/src/transaction/fees/zip317.rs#L135>
+
+    let marginal_fee: Amount<NonNegative> = MARGINAL_FEE.try_into().expect("fits in amount");
+
+    let tx_in_total_size: usize = transaction
+        .inputs()
+        .iter()
+        .map(|input| input.zcash_serialized_size())
+        .sum();
+
+    let tx_out_total_size: usize = transaction
+        .outputs()
+        .iter()
+        .map(|output| output.zcash_serialized_size())
+        .sum();
+
+    let n_join_split = transaction.joinsplit_count();
+    let n_spends_sapling = transaction.sapling_spends_per_anchor().count();
+    let n_outputs_sapling = transaction.sapling_outputs().count();
+    let n_actions_orchard = transaction.orchard_actions().count();
+
+    let tx_in_logical_actions = div_ceil(tx_in_total_size, P2PKH_STANDARD_INPUT_SIZE);
+    let tx_out_logical_actions = div_ceil(tx_out_total_size, P2PKH_STANDARD_OUTPUT_SIZE);
+
+    let logical_actions = max(tx_in_logical_actions, tx_out_logical_actions)
```

Comment:
Optional: move logical actions calculation to an `impl Transaction { fn count_logical_actions(&self) }` next to this fn

Reply:
I'd prefer to just have `conventional_fee` as a field, there's no other uses for logical actions yet, and the serialization of transaction data can be slightly expensive due to cryptographic operations when serializing types.